tests/test_usecases.py

```diff
@@ -436,6 +473,392 @@ def test_disable_dataset_with_db(db, schema_data, dataset_data):
     assert status[0] == "DISABLED"
 
 
+@pytest.mark.database
+@pytest.mark.usefixtures("monetdb_container", "cleanup_db")
+def test_disable_dataset_already_disabled_with_db(db, schema_data, dataset_data):
+    InitDB(db).execute()
+    AddSchema(db).execute(schema_data)
+    AddDataset(db).execute(dataset_data, "schema", "1.0")
+    datasets = db.get_datasets()
+    status = db.execute(f"SELECT status FROM mipdb_metadata.datasets").fetchone()
+    assert status[0] == "DISABLED"
+
+    with pytest.raises(UserInputError):
+        DisableDataset(db).execute(
+            datasets[0], schema_data["code"], schema_data["version"]
+        )
+
+
+def test_tag_schema():
+    db = MonetDBMock()
+    code = "schema"
+    version = "1.0"
+    tag = "tag"
+    key_value = ("key", "value")
+    add = True
+    remove = False
+    TagSchema(db).execute(code, version, tag, key_value, add, remove)
+    assert "UPDATE mipdb_metadata.schemas SET properties" in db.captured_queries[0]
+    assert "Sequence('action_id_seq'" in db.captured_queries[1]
+    assert 'INSERT INTO "mipdb_metadata".actions ' in db.captured_queries[2]
+
+
+@pytest.mark.database
+@pytest.mark.usefixtures("monetdb_container", "cleanup_db")
+def test_tag_schema_tag_or_key_value_needed_with_db(db, schema_data):
+    # Setup
+    InitDB(db).execute()
+    AddSchema(db).execute(schema_data)
+
+    # Test
+    TagSchema(db).execute(
+        schema_data["code"], schema_data["version"], "tag", ("key", "value"), True, False
```

Comment:
In a function with 6 arguments you **have** to pass them as keyword args, otherwise no one knows what's going on. But again, this is related to the fact that your usecase does 4 different things.